docs/architecture/adr-063-core-module-api.md

```diff
@@ -0,0 +1,546 @@
+# ADR 063: Core Module API
+
+## Changelog
+
+* 2022-08-18 First Draft
+* 2022-12-08 First Draft
+
+## Status
+
+PROPOSED Not Implemented
+
+## Abstract
+
+A new core API is proposed as a way to develop cosmos-sdk applications that will eventually replace the existing
+`AppModule` and `sdk.Context` frameworks a set of core services and extension interfaces. This core API aims to:
+- be simpler,
+- more extensible,
+- more stable than the current framework,
+- enable deterministic events and queries, 
+- support event listeners and
+- [ADR 033: Protobuf-based Inter-Module Communication](./adr-033-protobuf-inter-module-comm.md) clients.
+
+## Context
+
+Historically modules have exposed their functionality to the framework via the `AppModule` and `AppModuleBasic`
+interfaces which have the following shortcomings:
+* both `AppModule` and `AppModuleBasic` need to be defined and registered which is counter-intuitive
+* apps need to implement the full interfaces, even parts they don't need (although there are workarounds for this),
+* interface methods depend heavily on unstable third party dependencies, in particular Tendermint,
+* legacy required methods have littered these interfaces for far too long
+
+In order to interact with the state machine, modules have needed to do a combination of these things:
+* get store keys from the app
+* call methods on `sdk.Context` which contains more or less the full set of capability available to modules.
+
+By isolating all the state machine functionality into `sdk.Context`, the set of functionalities available to
+modules are tightly coupled to this type. If there are changes to upstream dependencies (such as Tendermint)
+or new functionalities are desired (such as alternate store types), the changes need impact `sdk.Context` and all
+consumers of it (basically all modules). Also, all modules now receive `context.Context` and need to convert these
+to `sdk.Context`'s with a non-ergonomic unwrapping function.
+
+Any breaking changes to these interfaces, such as ones imposed by third-party dependencies like Tendermint, have the
+side effect of forcing all modules in the ecosystem to update in lock-step. This means it is almost impossible to have
+a version of the module which can be run with 2 or 3 different versions of the SDK or 2 or 3 different versions of
+another module. This lock-step coupling slows down overall development within the ecosystem and causes updates to
+components to be delayed longer than they would if things were more stable and loosely coupled.
+
```

Comment:
can we add an alternative section to this? its kind of hard to gauge design when the decision is the only consideration